config/mfDefsGEM.mk

```diff
@@ -100,13 +105,11 @@ CLANG60Flags = -std=c++17
 CLANG70Flags = -std=c++17
 
 # User C and CC flags
-UserCFlags =-DGIT_VERSION=\"$(GIT_VERSION)\" -DGEMDEVELOPER=\"$(GEMDEVELOPER)\"
+UserCFlags =-DGIT_VERSION=\"$(GIT_VERSION)\" -DGEMDEVELOPER=\"$(GEMDEVELOPER)\" -DXDAQ15
 
 ## want to pick the best options based on gcc/clang version
-# CLANGNUM := $(shell clang -dumpversion|sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g' -e 's/^[0-9]\{3,4\}$/&00/')
-# GCCNUM   := $(shell gcc   -dumpversion|sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g' -e 's/^[0-9]\{3,4\}$/&00/')
-CLANGNUM := $(shell clang -dumpversion|sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g')
-GCCNUM   := $(shell gcc   -dumpversion|sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g')
+CLANGNUM:= $(shell clang -dumpfullversion -dumpversion|sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g')
+GCCNUM  := $(shell gcc   -dumpfullversion -dumpversion|sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g')
```

Comment:
I had to add `dumpfullversion` for the `devtoolset` `gcc` 8 compiler to print anything other than the major version.
I tested with various compilers and this combination seemed to get the desired behaviour for all of them

Reply:
Indeed, the `dumpfullversion` flag should be used for `gcc` according to the man man page:
>        -dumpversion
>           Print the compiler version (for example, 3.0, 6.3.0 or 7)---and don't do anything else.  This is the compiler version used in filesystem paths and specs.
>           Depending on how the compiler has been configured it can be just a single number (major version), two numbers separated by a dot (major and minor version) or
>           three numbers separated by dots (major, minor and patchlevel version).
>
>       -dumpfullversion
>           Print the full compiler version---and don't do anything else. The output is always three numbers separated by dots, major, minor and patchlevel version.

However, this flag [does not exist for `clang`](https://clang.llvm.org/docs/ClangCommandLineReference.html):
``` bash
[laurent@laurent-laptop ~]$ clang -dumpfullversion
clang-9: error: no input files
```

Where I was wrong is that the first option is used and not the last one (looks like to be an exception with the `dumpversion` and `dumpfullversion` flags).